backend/src/junior/interceptors/edit.interceptor.ts

```diff
@@ -21,6 +22,8 @@ export class JuniorEditInterceptor implements NestInterceptor {
         body.phoneNumber ? dataChanged = dataChanged || body.phoneNumber !== userToEdit.phoneNumber : body.phoneNumber = userToEdit.phoneNumber;
         body.firstName ? dataChanged = dataChanged || body.firstName !== userToEdit.firstName : body.firstName = userToEdit.firstName;
         body.lastName ? dataChanged = dataChanged || body.lastName !== userToEdit.lastName : body.lastName = userToEdit.lastName;
+        // Additional check here as the ORM seems not to be checking this on update.
+        if (!(new Validator()).isPhoneNumber(body.phoneNumber, 'FI')) { throw new BadRequestException(content.DataIncorrect); }
```

Comment:
This can be done in the entity itself, see: https://github.com/typeorm/typeorm/issues/913

Example:

```
import {
  Entity, PrimaryGeneratedColumn, Column, BaseEntity, BeforeInsert, BeforeUpdate,
} from 'typeorm';
import { validateOrReject, IsDefined } from 'class-validator';

@Entity()
export class Category extends BaseEntity {
  @PrimaryGeneratedColumn('uuid')
  public id: string;

  @Column({ nullable: false })
  @IsDefined()
  public name: string;

  // HOOKS
  @BeforeInsert()
  @BeforeUpdate()
  async validate() {
    await validateOrReject(this);
  }
}
```

Dunno if it's the best solution though.

Reply:
I had tried this during development; however, the code did not trigger. I can re-investigate incase this was a caused by a coding issue.